packages/shared_preferences/shared_preferences_ios/pigeons/messages.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:pigeon/pigeon.dart';
+
+@ConfigurePigeon(PigeonOptions(
+  dartOut: 'lib/messages.g.dart',
+  objcHeaderOut: 'ios/Classes/messages.g.h',
+  objcSourceOut: 'ios/Classes/messages.g.m',
+  copyrightHeader: 'pigeons/copyright_header.txt',
+))
+@HostApi()
+abstract class SharedPreferencesApi {
+  bool remove(String key);
+  bool setBool(String key, bool value);
+  bool setDouble(String key, double value);
+  bool setInt(String key, int value);
+  bool setString(String key, String value);
+  bool setStringList(String key, List<String> value);
+  bool clear();
```

Comment:
I don't think so, the platform_interface states that a bool should be returned from these methods.  It should be the host platform that decides the value, not the glue code between the platform_interface and the host code.

Reply:
> I don't think so, the platform_interface states that a bool should be returned from these methods.

I'm not suggesting that the implementation of the platform interface—which is Dart code—violate that. This structure is not the implementation of the platform interface, it's a completely internal API boundary.

> It should be the host platform that decides the value, not the glue code between the platform_interface and the host code.

shared_preferences_ios is "the host platform". It's all iOS code. Which parts of that iOS code are written in Dart and which parts in ObjC are up to us, as is defining the API between those layers.

There is no reason to put useless return value on an internal API boundary.

Reply:
> There is no reason to put useless return value on an internal API boundary.

Here's how you are looking at it:

`(dart_platform_interface)->(dart_host_implementation)->(objc_host_implementation)`

Here's how I think you should be thinking about it:

`(dart_platform_interface)->(dart_glue)->(host_implementation)`

It makes maintaining much easier since logic for implementing `dart_platform_interface` isn't spread across multiple files.

Reply:
> > There is no reason to put useless return value on an internal API boundary.
> 
> Here's how you are looking at it:
> 
> `(dart_platform_interface)->(dart_host_implementation)->(objc_host_implementation)`

Not really, no.

I look it at as:
```
Dart platform interface
Adapter code
System API
```

More specifically:
```
Dart platform interface
Adapter code (Dart)
Method channel/pigeon API
Adapter code (Native language)
System API
```

> Here's how I think you should be thinking about it:
> 
> `(dart_platform_interface)->(dart_glue)->(host_implementation)`

To the extent that I think we should have an abstract goal of minimizing the amount of adapter code on one side of the language boundary, I think it should be the native side, favoring writing adapter code in Dart, for the reasons laid out in https://docs.flutter.dev/go/platform-channels-in-federated-plugins (search for "Allows moving adapter code from native code to Dart")

Reply:
Well it doesn't help that what those return values mean is never defined, so I can't even tell you if the values are correct or if they need to be generated from the native code.  Imagine those return values actually meant something, they would have to be forwarded from the host platform to the plugin.  It seems a bit silly to make a special case about where the return values come from in the case where return values mean nothing.  I think it's just setting up the wrong example.

Reply:
> Well it doesn't help that what those return values mean is never defined

On Android the setters can fail, and the return values is whether setting succeeded.

A PR fixing the docs to say that sounds like a good idea.

> so I can't even tell you if the values are correct

Seeing a value can't fail on iOS, so unconditionally returning true is the correct implementation on iOS.

> or if they need to be generated from the native code.

We know that they don't, because we know what the APIs on iOS are.

> Imagine those return values actually meant something, they would have to be forwarded from the host platform to the plugin.

If there were meaningful return values that we needed from the native side on iOS, I wouldn't have made that review comment in the first place.

> It seems a bit silly to make a special case about where the return values come from in the case where return values mean nothing.

There is, and will only ever be, one implementation of the API you are defining with this pigeon file. When there is only one case, "special case" isn't meaningful.

> I think it's just setting up the wrong example.

This isn't an example. It's a specific, concrete case.